### DISCUSSION TITLE
Issue running .NET/Emgu.CV code on Raspberry Pi

### DISCUSSION THREAD

**Discussion by davefloodnz** [emgucv/emgucv#912] [Q&A]
Hi. Please excuse my noob question; I'm new to Raspberry Pi, Linux & cross-platform dev.

I have written a small .NET 8 console app in Visual Studio which uses EMGU.CV to access camera. It works on my PC.

I published with target Runtime set to Linux-Arm.

I copied output to my Raspberry Pi 4 (running 32 bit Debian Bullseye). but when I run it crashes once the code uses EMGU (Creating a new Mat() object)

 ---> System.TypeInitializationException: The type initializer for 'Emgu.CV.Mat Invoke' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'Emgu.CV.CvInvoke' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'cvextern' or one of its dependencies. In order to help diagnose loading problems, consider using a tool like strace. If you're using glibc, consider setting the LD_DEBUG environment variable:
/home/dave/DeleteArmPublishTest/cvextern.so: cannot open shared object file: No  such file or directory
libgeotiff.so.5: cannot open shared object file: No such file or directory
libcvextern.so: cannot open shared object file: No such file or directory
/home/dave/DeleteArmPublishTest/cvextern: cannot open shared object file: No such file or directory
/home/dave/DeleteArmPublishTest/libcvextern: cannot open shared object file: No such file or directory

I assume this issue is because OpenCV or Emgu or both are missing on the Pi.

I have the Emgu.CV.runtime.debian-arm package(4.8.0.5324) included in project along with the Windows runtime, but I'm not sure what I'm supposed to do with that? 

- Does the Emgu.CV.runtime.debian-arm package include all the required EMGU & OpenCV stuff and I have missed a step including it?
- Do I need to be installing EMGU (or even OpenCV) directly on Raspberry Pi?
- Do I need to be building something directly on the Pi like some posts mention or is that a thing of the past?

Any suggestion?

**Comment by jay8ee** [comment]
You need to install the dependencies on the PI.

```
sudo apt update
sudo apt full-upgrade
sudo apt install libgeotiff5
sudo apt install libvtk9.1
```
